src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -644,43 +644,58 @@ void GenerateFieldAccessor(const FieldDescriptor* field, const Options& options,
   // Generate getter.
   GenerateFieldDocComment(printer, field, options, kFieldGetter);
 
+  // deprecation
+  std::string deprecation_std_string = (field->options().deprecated()) ? "@trigger_error('" +
```

Comment:
Can you confirm that this outputs a warning only, and won't halt execution of the script?

Reply:
Yes, I can confirm it. That's why the `@` character is added. This way, you can opt in to handle deprecations, compare the [current Symfony LTS documentation](https://symfony.com/doc/4.4/contributing/code/conventions.html#deprecating-code).